Simplified/Accounts/AgeCheck/NYPLAgeCheck.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+
+protocol NYPLAgeCheckValidationDelegate: class {
+  var minYear : Int { get }
+  var currentYear : Int { get }
+  var birthYearList : [Int] { get }
+  var ageCheckCompleted : Bool { get set }
+  
+  func isValid(birthYear: Int) -> Bool
+  
+  func ageCheckCompleted(_ birthYear: Int)
+  func ageCheckFailed()
+}
+
+@objc protocol NYPLAgeCheckVerifying {
+  func verifyCurrentAccountAgeRequirement(userAccountProvider: NYPLUserAccountProvider,
+                                          currentLibraryAccountProvider: NYPLCurrentLibraryAccountProvider,
+                                          completion: ((Bool) -> ())?) -> Void
+}
+
+@objc protocol NYPLAgeCheckChoiceStorage {
+  var userPresentedAgeCheck: Bool { get set }
+}
+
+@objcMembers final class NYPLAgeCheck : NSObject, NYPLAgeCheckValidationDelegate, NYPLAgeCheckVerifying {
+  
+  // Members
+  let serialQueue = DispatchQueue(label: "\(Bundle.main.bundleIdentifier!).ageCheck")
+  var handlerList = [((Bool) -> ())]()
+  var isPresenting = false
+  var ageCheckCompleted: Bool = false
+  let ageCheckChoiceStorage: NYPLAgeCheckChoiceStorage
+  
+  var minYear: Int = 1900
+  
+  var currentYear: Int = Calendar.current.component(.year, from: Date())
```

Comment:
since minYear, currentYear are get only and we're not using them as computed properties, and birthYear does not change at all with a change in state, does it make more sense to do this instead?
`// NYPLAgeCheckDelegate
    let minYear: Int
    let currentYear: Int
    let birthYearList: [Int]

    init(ageCheckChoiceStorage: NYPLAgeCheckChoiceStorage) {
        self.ageCheckChoiceStorage = ageCheckChoiceStorage
        minYear = 1900
        currentYear = Calendar.current.component(.year, from: Date())
        birthYearList = Array(minYear...currentYear)
        super.init()
    }`

Even though the compiler is smart enough to know that a variable is actually a constant, it kinda makes it easier on the eyes. Just a _weak_ suggestion.

Reply:
those should all be `lets` if they never change, that's definitely a good suggestion -- nice catch. 

Initializing them in the initializer vs inline is usually cosmetic, but in this case it allows for clearer behavior for `birthYearList` so I'm in favor of this.